sherpa-onnx/csrc/provider-config.cc

```diff
@@ -113,9 +113,11 @@ void ProviderConfig::Register(ParseOptions *po) {
 }
 
 bool ProviderConfig::Validate() const {
-  if (device < 0) {
-    SHERPA_ONNX_LOGE("device: '%d' is invalid.", device);
-    return false;
+  if(provider == "cuda" || provider == "trt") {
```

Comment:
I mean, do you need to implement:
```
if provider == "cuda" and not cuda_config.Validate():
  return false

if provider == 'trt' and not trt_config.Validate():
  return false
```

(Note it's just pseudo code. I think you get the ideas.)

Reply:
oh yes sorry,
literally I am not thinking through anymore 😄 , please don't mind
pybind took a good score on me 😮‍💨